client/src/modules/stock/entry/modals/lots.modal.js

```diff
@@ -9,196 +9,142 @@ StockDefineLotsModalController.$inject = [
+  vm.checkAll = checkAll;
   vm.removeLot = removeLot;
-  vm.handleChange = handleChange;
-  vm.onDateChangeCallback = onDateChangeCallback;
 
-  // init
   function init() {
-    vm.inventory = Data.inventory;
-    vm.remainingQuantity = vm.inventory.quantity;
-
-    if (vm.inventory.lots.length) {
-      vm.gridOptions.data = vm.inventory.lots;
-    } else {
+    if (!vm.stockLine.lots.length) {
       addLot();
+    } else {
+      // if there is at least one lot already, then check only
+      checkAll();
     }
   }
 
-  // add lot
   function addLot() {
-    if (vm.remainingQuantity <= 0 && vm.entryType !== 'integration') {
-      vm.maxLotReached = true;
-      return;
-    }
-    vm.gridOptions.data.push({
-      is_valid        : false,
-      lot             : vm.inventory.lot || '',
-      expiration_date : vm.inventory.expiration_date ? new Date(vm.inventory.expiration_date) : new Date(),
-      quantity        : vm.remainingQuantity,
+    vm.stockLine.lots.push({
+      isValid: false,
+      lot: null,
+      expiration_date: new Date(),
+      quantity: 0,
     });
 
-    //  if it is a transfer reception, so force the validation on the single element
-    if(vm.entryType === 'transfer_reception') {
-      handleChange(vm.gridOptions.data[0]);
-    }
-
-  }
-
-  // remove lot
-  function removeLot(index) {
-    vm.gridOptions.data.splice(index, 1);
+    checkAll();
   }
 
-  // handle date change
-  function onDateChangeCallback(date, inventory) {
-    inventory.expiration_date = date;
-    // notify date change
-    handleChange(inventory);
-  }
-
-
-  function hasInvalidInventory(hasQuantity, hasExpiration) {
-    var error;
-
-    var isExpiration = hasExpiration ? false : true;
-    var isExcessiveQuatity = hasQuantity ? false : true;
-
-    if (isExcessiveQuatity) {
-      vm.errorText = 'STOCK.ERRORS.EXCESSIVE_QUANTITY';
-      error = true;
-    } else if (isExpiration) {
-      vm.errorText = 'STOCK.ERRORS.PLEASE_CHECK_EXPIRY_DATE';
-      error = true;
-    }
-
-    return error;
+  function onRegisterApi(api) {
+    vm.gridApi = api;
   }
 
-  // handleChange
-  function handleChange(inventory) {
-    vm.error =  false;
-
-    var sum = vm.gridOptions.data.reduce(sumQuantity, 0);  
-    var hasQuantity = (vm.inventory.quantity >= sum);
-    var hasLotLabel = inventory.lot;
-    var hasExpiration = (new Date(inventory.expiration_date) >= new Date());
-    inventory.is_valid = (hasQuantity && hasLotLabel && hasExpiration);
-
-    vm.error = hasInvalidInventory(hasQuantity, hasExpiration);
-    vm.submitError = vm.error ? vm.error : false;
+  function checkAll() {
+    vm.isLotExist = vm.stockLine.lots.length > 0;
 
+    vm.isEveryLineCorrect = vm.stockLine.lots.every(function (lot) {
+      return lot.isValid === true;
+    });
 
-    vm.remainingQuantity = (vm.inventory.quantity - sum >= 0) ? vm.inventory.quantity - sum : 0;
-    vm.sum = sum;
+    vm.sum = vm.stockLine.lots.reduce(function (x, y) {
```

Comment:
Depending how often this is called (it looks like it might be called a lot), you could move this function out of the `checkAll()` function.  That way the V8 engine doesn't keep allocating a new function and discarding it when `checkAll()` completes.  Like this:
```js
function summer(x, y) {
  return x + y.quantity
}

function checkAll() {
  vm.sum = vm.stockLine.lots.reduce(summer, 0);

   // etc...
}